header/sync/sync.go

```diff
@@ -88,6 +88,23 @@ func (s *Syncer) WaitSync(ctx context.Context) error {
 	return err
 }
 
+// Head returns the Syncer's pending head if it exists, and if not, eagerly
+// fetches the head from the header.Exchange.
+func (s *Syncer) Head(ctx context.Context) (*header.ExtendedHeader, error) {
+	// try pending head
+	pendHead := s.pending.Head()
+	if pendHead != nil {
+		return pendHead, nil
+	}
+	// if no pending head, eagerly fetch head from network
+	objHead, err := s.exchange.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+	s.pending.Add(objHead)
+	return objHead, nil
+}
```

Comment:
This should work; however: 
* we should think about not making Head requests via Exchange in a happy case when pending is empty and the node is fully synced.
	* When the node is fully synced, new gossiped headers are written directly into the store, avoiding the pending cache.
* We might need to trigger sync if we added a new header to the pending cache so that node does not wait for gossiped headers, as it has a sync target already

Reply:
Let's sync on those